libs/renterd-types/src/utils.ts

```diff
@@ -0,0 +1,3 @@
+export type DeepPartial<T> = {
+  [P in keyof T]?: T[P] extends object | undefined ? DeepPartial<T[P]> : T[P]
```

Comment:
If I'm understanding the approach here correctly, undefined may be unnecessary. You might throw a test at a version of this without `| undefined` that has undefined values and see if they're treated the way you want.

The other thing that comes to mind is values that are objects but may not be obvious: functions and Date stuff--those probably wont come through here--but what about arrays? Those will hit the recursive case here, I believe, and that may result in some funkiness.

Otherwise, LGTM!